src/libraries/Microsoft.PowerFx.Core/Texl/Intellisense/IntellisenseData/IntellisenseData.cs

```diff
@@ -265,7 +270,72 @@ internal virtual void BeforeAddSuggestionsForDottedNameNode(TexlNode node)
         /// <returns>
         /// True if the function may be suggested, false otherwise.
         /// </returns>
-        internal virtual bool ShouldSuggestFunction(TexlFunction function) => true;
+        internal virtual bool ShouldSuggestFunction(TexlFunction function)
+        {
+            if (CurFunc == null)
+            {
+                return true;
+            }
+
+            DType expectedReturnType;
+
+            expectedReturnType = GetCurrentArgType();
```

Comment:
>expectedReturnType [](http://example.com/codeflow?start=12&length=18)

I am little confused by this name. Why is the name of return type from GetCurrentArgType() expectedReturnType?